### ISSUE TITLE
Support to build as shared library

### ISSUE THREAD

**Issue by topazus** [YACLib/YACLib#245]

**Comment by MBkkt** [comment]
Is it make sense?

**Comment by topazus** [comment]
Yes, I think it is proper to support to build shared or static library. The CMake variable `BUILD_SHARED_LIBS` can be used as the option to choose to build shared or static. Also, set soversion when building as shared library is also suggested, which can help to indicate the ABI/API change.

Ref:
https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html
https://cmake.org/cmake/help/latest/prop_tgt/SOVERSION.html
https://github.com/simdjson/simdjson/issues/52

**Comment by MBkkt** [comment]
I don't think we have stable ABI.

In general library is really tiny and it doesn't feel like shared library is good option for it.

Also with dynamic linking it's impossible to rely on lto.
We rely on lto to decrease compile time and move more code out of headers file.
